packages/framework/src/Component/Translation/FrontendApiDetectorTrait.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Translation;
+
+trait FrontendApiDetectorTrait
+{
+    protected bool $isFrontendApiBundleConstraintClass = false;
+
+    /**
+     * @param string $className
+     */
+    protected function setIfFrontendApiClass(string $className): void
+    {
+        if (str_starts_with($className, 'Shopsys\\FrontendApiBundle\\') || str_starts_with($className, 'App\\FrontendApi\\')) {
+            $this->isFrontendApiBundleConstraintClass = true;
+        }
+    }
+
+    /**
+     * @return string
+     */
+    protected function getTranslationDomain(): string
+    {
+        if ($this->isFrontendApiBundleConstraintClass) {
+            return Translator::CUSTOMER_VALIDATOR_TRANSLATION_DOMAIN;
+        }
```

Comment:
maybe we could come up with more specific name for the trait? From the current name, it is not obvious that CUSTOMER_VALIDATOR_TRANSLATION_DOMAIN will be used if in FE API...I am afraid it might be misued. What about `FrontendApiValidationTranslationDomainDetectorTrait`? Or something like that

Reply:
renamed in 9df02a2d34